app/controllers/coronavirus_form/essential_supplies_controller.rb

```diff
@@ -18,10 +18,13 @@ def submit
       respond_to do |format|
         format.html { render controller_path, status: :unprocessable_entity }
       end
+    elsif session[:check_answers_seen] && session[:dietary_requirements].present? && session[:carry_supplies].present?
+      update_session_store
+      redirect_to check_your_answers_url
     elsif @form_responses[:essential_supplies] == I18n.t("coronavirus_form.questions.essential_supplies.options.option_no.label")
       update_session_store
       redirect_to dietary_requirements_url
-    elsif session[:check_answers_seen]
+    elsif session[:check_answers_seen] && @form_responses[:essential_supplies] == I18n.t("coronavirus_form.questions.essential_supplies.options.option_yes.label")
```

Comment:
Thanks for fixing the logic. It has made the if/else a bit unwieldy though. Perhaps, rather than having this if/else, the logic for figuring out the next page could be moved to a method?

e.g.

```
def next_page_url
    return dietary_requirements_url if answer_next_question?
    return check_your_answers_url if session[:check_answers_seen]

    basic_care_needs_url
  end

  def answer_next_question?
    @form_responses[:essential_supplies] == I18n.t("coronavirus_form.questions.essential_supplies.options.option_no.label") &&
      (session[:dietary_requirements].blank? || session[:carry_supplies].blank?)
  end
```

And then the routing logic could be simplified to:

```
if invalid_fields.any?
      flash.now[:validation] = invalid_fields
      log_validation_error(invalid_fields)

      respond_to do |format|
        format.html { render controller_path, status: :unprocessable_entity }
      end
    else
      update_session_store
      redirect_to next_page_url
    end
```

Reply:
Nice idea, have made this change.